gp2gp-translator/src/test/java/uk/nhs/adaptors/pss/translator/mapper/ImmunizationMapperTest.java

```diff
@@ -183,6 +205,90 @@ public void mapObservationToImmunizationWithUNKVaccineCode() {
         assertImmunizationWithDefaultVaccineCode(immunization);
     }
 
+    @Test
+    public void When_EhrCompositionWithNoPatConfidentialityCode_Expect_MetaFromConfidentialityServiceWithSecurity() {
+        final var metaWithSecurity = MetaFactory.getMetaFor(META_WITH_SECURITY, META_PROFILE);
+        when(
+            confidentialityService.createMetaAndAddSecurityIfConfidentialityCodesPresent(
+                eq("Immunization-1"),
+                confidentialityCodeCaptor.capture(),
+                confidentialityCodeCaptor.capture()
+            )
+        ).thenReturn(metaWithSecurity);
+        final var ehrExtract = unmarshallEhrExtract("immunization_with_ehrComposition_nopat_confidentiality_code.xml");
+
+        List<Immunization> immunizations = immunizationMapper.mapResources(
+            ehrExtract, getPatient(),
+            getEncounterList(), PRACTISE_CODE
+        );
+        var securityMeta = immunizations
+            .get(0)
+            .getMeta()
+            .getSecurity(NOPAT_URL_CODESYSTEM, NOPAT_CODE);
+
+        assertAll(
+            () -> {
+                assertThat(confidentialityCodeCaptor.getAllValues().get(0).isPresent())
+                    .isTrue();
+                assertThat(confidentialityCodeCaptor.getAllValues().get(0).get())
+                    .usingRecursiveComparison()
+                    .isEqualTo(getNoPatCV());
+            },
+            () -> {
+                assertThat(securityMeta).
+                    isNotNull();
+                assertThat(securityMeta.getDisplay())
+                    .isEqualTo(NOPAT_DISPLAY);
+            }
+        );
+    }
+
+    @Test
+    public void When_ObservationWithNoPatConfidentialityCode_Expect_MetaFromConfidentialityServiceWithSecurity() {
+        final var metaWithSecurity = MetaFactory.getMetaFor(META_WITH_SECURITY, META_PROFILE);
+        when(
+            confidentialityService.createMetaAndAddSecurityIfConfidentialityCodesPresent(
+                eq("Immunization-1"),
+                confidentialityCodeCaptor.capture(),
+                confidentialityCodeCaptor.capture()
+            )
+        ).thenReturn(metaWithSecurity);
+        final var ehrExtract = unmarshallEhrExtract("immunization_with_observation_nopat_confidentiality_code.xml");
+
+        List<Immunization> immunizations = immunizationMapper.mapResources(
+            ehrExtract, getPatient(),
+            getEncounterList(), PRACTISE_CODE
+        );
+        var securityMeta = immunizations
+            .get(0)
+            .getMeta()
+            .getSecurity(NOPAT_URL_CODESYSTEM, NOPAT_CODE);
+
+        assertAll(
+            () -> {
+                assertThat(confidentialityCodeCaptor.getAllValues().get(1).isPresent())
+                    .isTrue();
```

Comment:
I think this can be shortened!

```suggestion
                assertThat(confidentialityCodeCaptor.getAllValues().get(1))
                    .isPresent();
```

Reply:
Ahh yeah, I had that originally but unfortunately it introduces a warning as the compiler doesn't think that that asertThat(...).isPresent() is the same as an isPresent() check!

![image](https://github.com/user-attachments/assets/c682390e-a6b8-445a-8379-0e0a1a4533bb)

Reply:
Scrap this, refactoring to extract the variable addressed the issue